astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -333,24 +333,25 @@ def test_invalid_parameters():
     """
     int_converter = ascii.convert_numpy(np.uint)
     converters = dict((i + 1, ascii.convert_numpy(np.uint)) for i in range(3))
-    invalid_params = {'delimiter': ',,',  # multi-char delimiter
-                      'comment': '##',  # multi-char comment
-                      'data_start': None,  # data_start=None
-                      'quotechar': '##',  # multi-char quote signifier
-                      'data_start': -1,  # negative data_start
-                      'header_start': -1,  # negative header_start
-                      'converters': converters,  # passing converters
-                      'Inputter': ascii.ContinuationLinesInputter,  # passing Inputter
-                      'header_Splitter': ascii.DefaultSplitter,  # passing Splitter
-                      'data_Splitter': ascii.DefaultSplitter
-                      }
-    for key, val in invalid_params.items():
-        with pytest.raises(ParameterError):
-            print('Trying {0}={1} using constructor'.format(key, val))
-            table = FastBasic(**{key: val}).read('1 2 3\n4 5 6')
-        with pytest.raises(ParameterError):
-            print('Trying {0}={1} using ascii.read'.format(key, val))
-            table = ascii.read('1 2 3\n4 5 6', format='fast_basic', guess=False, **{key: val})
+    invalid_params = {
+        'delimiter': [',,'],  # multi-char delimiter
+        'comment': ['##'],  # multi-char comment
+        'data_start': [None, -1],  # data_start=None and negative
+        'quotechar': ['##'],  # multi-char quote signifier
+        'header_start': [-1],  # negative header_start
+        'converters': [converters],  # passing converters
+        'Inputter': [ascii.ContinuationLinesInputter],  # passing Inputter
+        'header_Splitter': [ascii.DefaultSplitter],  # passing Splitter
+        'data_Splitter': [ascii.DefaultSplitter]
+        }
+    for key, values in invalid_params.items():
+        for val in values:
+            with pytest.raises(ParameterError):
+                print('Trying {0}={1} using constructor'.format(key, val))
```

Comment:
Anyone knows why it `print`s here?

Reply:
probably helped debugging at the time of writing? Was added in https://github.com/astropy/astropy/pull/2716 and hasn't been touched since.

Reply:
hm, okay. I wasn't sure if I should've removed them but if they could be useful it's probably better to keep them in.

Reply:
okay, I just checked and there are some additional `print`s in other test functions. So it's probably out of scope of this PR to evaluate their usefulness.

Reply:
I think the point of print statements like these is that, if an actual error occurs, the pytest output will list what was printed, and thus you know for what entry the test failed. In principle, I think the more proper way to do this is to use `pytest.mark.parametrize` on a smaller test function.

Reply:
Yes, I think @mhvk got it right.  I would just leave these for now.